File: exercises/ch1/ba1c.ts
function reverseComplement(pattern: string): string {
    return pattern
        .split('')
        .map((n: string) => {
            switch (n) {
                case 'A':
                    return 'T';
                case 'C':
                    return 'G';
                case 'T':
                    return 'A';
                case 'G':
                    return 'C';
                default:
                    throw new Error(`Invalid nucleotide ${n}`);
            }
        })
        .reverse()
        .join('');
}

export default function ba1c(params: string[]) {
    const [pattern] = params;
    return reverseComplement(pattern);
}

File: exercises/ba1c.py
from library.algorithms import reverse_complement


def ba1c(params: list[str]):
    pattern = params[0]
    return reverse_complement(pattern)


def test_ba1c(benchmark):
    expected = "ACCGGGTTTT"
    result = benchmark(ba1c, ["AAAACCCGGT"])
    assert expected == result
